File: AoC-2024/Day-04/2024-04-2.py
with open('Input.txt', 'r') as f:
    txt = f.read()

M = txt.split('\n')
R,C = len(M),len(M[0])

ans = 0
for r in range(1,R-1):
    for c in range(1,C-1):
        if M[r][c] != 'A':
            continue
        if sorted([M[r-1][c-1],M[r+1][c+1]]) != ['M','S']:
            continue
        if sorted([M[r-1][c+1],M[r+1][c-1]]) != ['M','S']:
            continue
        ans += 1

print(ans)

File: AoC-2024/Day-04/2024-04-2.js
const fs = require('fs')
let txt = fs.readFileSync('Input.txt', 'utf8')

let M = txt.split('\n')
let R = M.length
let C = M[0].length

let ans = 0
for (let r = 1; r < R-1; r++) {
    for (let c = 1; c < C-1; c++) {
        if (M[r][c] != 'A') {
            continue
        }
        let diag0 = M[r-1][c-1]+M[r+1][c+1]
        let diag1 = M[r-1][c+1]+M[r+1][c-1]
        if (diag0 != 'MS' && diag0 != 'SM') {
            continue
        }
        if (diag1 != 'MS' && diag1 != 'SM') {
            continue
        }
        ans += 1
    }
}

console.log(ans)
